source/common/http/utility.h

```diff
@@ -233,6 +233,46 @@ void transformUpgradeRequestFromH2toH1(HeaderMap& headers);
  */
 void transformUpgradeResponseFromH2toH1(HeaderMap& headers, absl::string_view upgrade);
 
+/**
+ * The non template implementation of resolvePerFilterConfig. see
+ * resolvePerFilterConfig for docs.
+ */
+const Router::RouteSpecificFilterConfig*
+resolvePerFilterConfigGeneric(const std::string& filter_name,
+                              const Router::RouteConstSharedPtr& route);
+
+/**
+ * Retreives the route specific config. Route specific config can be in a few
+ * places, that are checked in order. The first config found is returned. The
+ * order is:
+ * - the routeEntry() (for config that's applied on weighted clusters)
+ * - the route
+ * - and finally from the virtual host object (routeEntry()->virtualhost()).
```

Comment:
Thanks for adding this. I'm kind of surprised we didn't have this already. I guess the issue might be when you have multiple configs, should we have precedence or merge semantics? CC @rshriram for comment as well.

Seems you might actually want merge in your use case?

Reply:
Yes, this utility is used indeed to get the most specific route config.
Not sure why merge is needed in this case - I think it is getting the most specific is better for this case, especially for disable setting.

Reply:
Disable makes sense to be most specific. For the attribute context, a merge is more likely the right thing; you might have properties that apply at virtual host level that you want to append with properties at the individual route.

Reply:
sounds good - I will merge the context extensions in order.